test/extensions/filters/http/ext_proc/ordering_test.cc

```diff
@@ -887,6 +890,20 @@ TEST_F(FastFailOrderingTest, GrpcErrorOnStartRequestBody) {
   EXPECT_EQ(FilterDataStatus::StopIterationNoBuffer, filter_->decodeData(req_body, true));
 }
 
+// gRPC failure while opening stream with only request body enabled
+TEST_F(FastFailOrderingTest, GrpcErrorOnStartRequestBodyBufferedPartial) {
+  initialize([](ExternalProcessor& cfg) {
+    auto* pm = cfg.mutable_processing_mode();
+    pm->set_request_header_mode(ProcessingMode::SKIP);
+    pm->set_request_body_mode(ProcessingMode::BUFFERED_PARTIAL);
+  });
```

Comment:
I guess this isn't relevant to this PR but any reason why this needs to be a runtime failure vs a configuration validation?

Reply:
This is actually valid configuration -- the user elected to only send the request body but not the header. But this test is checking what happens specifically if there is a gRPC error in this case, because in theory we can open the gRPC stream in response to any phase in the lifecycle and we want to make sure that we test for that.